src/main/java/kaiyap/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package kaiyap;
+
+import java.time.LocalDateTime;
+
+public class Deadline extends Task {
+    private LocalDateTime deadline;
+    public Deadline(String listItem, String inputItem, LocalDateTime deadline) {
+        super(listItem, inputItem);
+        this.deadline = deadline;
+    }
+
+    public LocalDateTime getDeadline() {
+        return deadline;
+    }
+
+    public void setDeadline(LocalDateTime deadline) {
+        this.deadline = deadline;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" +
+                (this.completedTask ? "[X] " : "[ ] ") +
+                this.listItem +
+                " (by: " +
+                this.deadline + ")"
+                ;
+    }
+}
```

Comment:
I like the logic

Reply:
I truly agree

Reply:
Thank you! Your words of affirmation mean a lot to me.